### PR TITLE
Fix Package.swift diagnostics line off by 1 for tools versions <= 5.7

### PR DESCRIPTION
**Pull request by bripeticca**
Fixes issue #7688

SwiftPM was implicitly prepending the manifest with 'import Foundation' for toolchains using Swift toolchain versions 5.7 and under. The original file itself was never overwritten, thus the diagnostics being reported would always present on the next line.

This is a quick fix that appends the same line to the end of the manifest; doing so assures that the intended behaviour doesn't change, but allows for correct diagnostics to be reported.

### Motivation:

Diagnostics would present on the incorrect line for Package.swift files using tools version 5.7 and under.

### Modifications:

When evaluating the manifest, a temporary directory copies the contents of the manifest to a temporary file; rather than prepending the import to Foundation at the top of the file (causing all lines in the diagnostic reports to be off by one), append the import to the bottom of the temporary manifest when dealing with tools versions 5.7 and under.

### Result:

The intended behaviour that requires users to explicitly import Foundation doesn't change for Swift versions 5.8 and up; diagnostics reported for Package.swift files for those using swift <= 5.7 will now show on the correct line.

### CODE REVIEW

File: Sources/PackageLoading/ManifestLoader.swift [hunk 1]
```diff
@@ -17,10 +17,10 @@ import Foundation
 import PackageModel
 import SourceControl
 
-import class TSCBasic.BufferedOutputByteStream
-import struct TSCBasic.ByteString
 import class Basics.AsyncProcess
 import struct Basics.AsyncProcessResult
+import class TSCBasic.BufferedOutputByteStream
+import struct TSCBasic.ByteString
```
Comment (MaxDesiatov) [inline Sources/PackageLoading/ManifestLoader.swift:23]: It's unclear at a glance which parts of the diff actually resolve the issue. We generally prefer to keep formatting changes to separate PRs, especially those affecting the whole file.

**Comment by bripeticca** [comment]
@swift-ci please test

**Review by MaxDesiatov — APPROVED** [review]
Great find, thanks!
